analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -0,0 +1,280 @@
+import com.here.ort.model.Project
+import com.here.ort.model.ProjectAnalyzerResult
+import com.here.ort.model.RemoteArtifact
+import com.here.ort.model.Hash
+import com.here.ort.model.Scope
+import com.here.ort.model.VcsInfo
+import com.here.ort.model.config.AnalyzerConfiguration
+import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.model.jsonMapper
+import com.here.ort.utils.CommandLineTool
+import com.here.ort.utils.log
+import com.here.ort.utils.textValueOrEmpty
+
+import com.moandjiezana.toml.Toml
+
+import com.vdurmont.semver4j.Requirement
+
+import java.io.File
+import java.util.SortedSet
+
+class Cargo(
+    name: String,
+    analyzerRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analyzerRoot, analyzerConfig, repoConfig), CommandLineTool {
+    class Factory : AbstractPackageManagerFactory<Cargo>("Cargo") {
+        override val globsForDefinitionFiles = listOf("Cargo.toml")
+
+        override fun create(
+            analyzerRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = Cargo(managerName, analyzerRoot, analyzerConfig, repoConfig)
+    }
+
+    companion object {
+        private const val REQUIRED_CARGO_VERSION = "1.0.0"
+        private const val SCOPE_NAME_DEPENDENCIES = "dependencies"
+        private const val SCOPE_NAME_DEV_DEPENDENCIES = "devDependencies"
+    }
+
+    override fun command(workingDir: File?) = "cargo"
+
+    override fun getVersionRequirement(): Requirement = Requirement.buildStrict(REQUIRED_CARGO_VERSION)
+
+    private fun runMetadata(workingDir: File): String = run(workingDir, "metadata", "--format-version=1").stdout
+
+    private fun extractCargoId(node: JsonNode) = node["id"].textValue()!!
+
+    private fun extractPackageId(node: JsonNode) =
+        Identifier(
+            type = "Cargo",
+            namespace = "",
+            name = node["name"].textValueOrEmpty(),
+            version = node["version"].textValueOrEmpty()
+        )
+
+    private fun extractRepositoryUrl(node: JsonNode) = node["repository"]?.textValue()
+
+    private fun extractVcsInfo(node: JsonNode): VcsInfo {
+        val url = extractRepositoryUrl(node)
+        val type = url?.let { "git" } ?: "" // for now cargo supports only git
+        return VcsInfo(type, url.orEmpty(), revision = "")
+    }
+
+    private fun extractDeclaredLicenses(node: JsonNode) =
+        node["license"].textValueOrEmpty().split("/")
+            .map { it.trim() }
+            .filter { it.isNotEmpty() }
+            .toSortedSet()
+
+    private fun extractSourceArtifact(
+        node: JsonNode,
+        hashes: Map<String, String>
+    ): RemoteArtifact? {
+        if (node["source"].textValueOrEmpty() != "registry+https://github.com/rust-lang/crates.io-index") {
+            return null
+        }
+        val name = node["name"]?.textValue() ?: return null
```

Comment:
Nit: Blank line before this one. (We generally use a blank line to separate a following statement from a preceding closing block.)